poradnia/users/views.py

```diff
@@ -11,12 +14,24 @@
 from .utils import PermissionMixin
 
 
-class UserDetailView(PermissionMixin, DetailView):
+class UserDetailView(PermissionRequiredMixin, DetailView):
     model = User
-    # These next two lines tell the view to index lookups by username
     slug_field = "username"
     slug_url_kwarg = "username"
 
+    def handle_no_permission(self):
+        if self.request.user.is_anonymous():
+            return redirect('/konta/login/?next=%s' % self.request.path)
+        else:
+            return HttpResponseForbidden("403 - not authorized")
+
+    def has_permission(self):
+        user = self.request.user
+        if user.has_perm("users.can_view_other") or self.request.path.startswith("/uzytkownik/{}/".format(user.username)):
```

Comment:
Poza tym - o ile dobrze pamiętam, ale mogę się mylić - to pierwsze wywołanie ```has_perm``` powoduje zapytanie do bazy danych. W tym wypadku ono może się okazać zbędne, bo można drugi warunek wykonać porównując stringi bez sprawdzania. Mikrooptymalizacja, ale czemu nie zwrócić na to uwagi.